src/java/org/apache/cassandra/db/memtable/SkipListMemtable.java

```diff
@@ -172,28 +180,13 @@ public long partitionCount()
     }
 
     @Override
-    public long rowCount()
+    public long getEstimatedAverageRowSize()
     {
-        DataRange range = DataRange.allData(metadata().partitioner);
-        ColumnFilter columnFilter = ColumnFilter.allRegularColumnsBuilder(metadata(), true).build();
-        return rowCount(columnFilter, range);
+        if (estimatedAverageRowSize == null || currentOperations.get() > estimatedAverageRowSize.operations * 1.5)
```

Comment:
To keep the constant probability of hitting new data in our calculation. If we did a constant delta, then imagine you had 1 million rows in meltable, and then added only 10 new rows. How likely are you going to even hit one of those 10 new rows in the sample? Very unlikely. So we'd compute the same number again and again. However, if you have only 1 row, and add another one - this may impact the result severely.

Reply:
makes sense, thanks for your clarification